Replace proxy #10521 ke (#130)
uyuni-project/uyuni-docs: modules/installation/pages/proxy-setup.adoc

====

Here is a detailed procedure step-by-step:

[[proxy.migration3.replace]]
.Procedure: Replacing a Proxy and Keeping the Clients Registered
. Before starting the actual migration procedure, save the data from the old proxy, if needed.
Consider copying important or manually created data to a central place that can also be accessed by the new proxy.
. Shut down the proxy.
. Install a new {productname} Proxy{nbsp}{productnumber}, following xref:install-proxy-unified.adoc[Proxy Installation].
. In the {productname} {webui} select the newly installed {productname} Proxy and delete it from the systems list.
. In the {webui}, create a reactivation key for the old proxy system: On the System Details tab of the old proxy click [guimenu]``Reactivation``.
Click [guimenu]``Generate New Key``, and make a note of the new key, as you will need it later.
For more information about reactivation keys, see xref:reference:systems/system-details/sd-reactivation.adoc[Reactivation Keys].
. After the installation of the new proxy, you might also need to:
** Copy the centrally saved data to the new proxy system
** Install any other needed software

> ** Set up TFTP synchronization if the proxy is used for autoinstallation

Author: fixed with an external commit.

[IMPORTANT]
====
During the installation of the proxy, clients will not be able to reach the {productname} Server.

> After a proxy system has been deleted from the systems list, all clients connected to this proxy will be (incorrectly) listed as `directly connected` to the {productname} Server.

Reviewer: ```suggestion
After a proxy system has been deleted, the systems list shows all clients that were connected to the deleted proxy, as now being directly connected to the {productname} Server.
This is incorrect.
```

Author: Yes, thanks, this is better. But I fear it is still confusing.  I think this is only temporarily and it is only incorrectly displayed or communicated to the user. The configuration itself is correct, and this will be visible to the user after a certain period of time.

Do we need a note or different wording, @Loquacity ?

Reviewer: Yeah, maybe. What about this instead? It moves the important information first, adds "temporarily" to make it clear it's not permanent, and splits the sentence up.

```suggestion
After you have a deleted a proxy, the systems list can be temporarily incorrect.
All clients that were previously connected to the proxy will show as being directly connected to the server instead.
```

[[proxy.pxe.dhcp]]

> === Configure DHCP for PXE through {susemgrproxy}

Reviewer 2: For coherence, shouldn't this be `{productname} Proxy` as well?

{productname} uses Cobbler for client provisioning.
PXE (tftp) is installed and activated by default.
Clients must be able to find the PXE boot on the {productname} Proxy using DHCP.
Use this DHCP configuration for the zone which contains the clients to be provisioned:

----
next-server: <IP_Address_of_Proxy>
filename: "pxelinux.0"
----

[[replace-susemgrproxy]]

> == Replace a {susemgrproxy}

Reviewer 2: Same.

[[replace-susemgrproxy]]
== Replace a {susemgrproxy}

A proxy does not contain any information about the clients that are connected to it.
Therefore, a proxy can be replaced by a new one at any time.
The replacement proxy must have the same name and IP address as its predecessor.

> Leave the old proxy in place while you prepare the replacement.

Reviewer 3: Maybe it should be clarified what "leave in place" means. Of course it should be shut down because the new proxy will have the same IP address as the old one and having two machines with the same IP is not really fun :)

Author: I push a change for clarity.

Create a reactivation key for this system and then register the new proxy using the reactivation key.
If you do not use the reactivation key, you will need to re-register all the clients against the new proxy.

[[proxy.migration3.replace]]
.Procedure: Replacing a Proxy and Keeping the Clients Registered
. Before starting the actual migration procedure, save the data from the old proxy, if needed.
Consider copying important or manually created data to a central place that can also be accessed by the new proxy.
. Shut down the proxy.

> . Install a new {susemgrproxy}{nbsp}{productnumber}, following xref:install-proxy-unified.adoc[Proxy Installation].

Reviewer 2: Shouldn't we use here:
```
. Install a new proxy{nbsp}{productnumber}, following xref:install-proxy-uyuni.adoc[Uyuni Proxy Installation].
```
Just as you did for Uyuni?

Author: I do not think so.  I think the preferred way to install the SUMA Proxy is to use the Unified Installer.

. In the {productname} {webui} select the newly installed {susemgrproxy} and delete it from the systems list.
. In the {webui}, create a reactivation key for the old proxy system: On the System Details tab of the old proxy click [guimenu]``Reactivation``.

> Click [guimenu]``Generate New Key``, and make a note of the new key, as you will need it later.

Reviewer 3: In general you should be able to use the same key you also used for the previous proxy. There is no general need for a new key.

Author: ok, I'll add that clarification.
